core/templates/pages/admin-page/misc-tab/admin-misc-tab.directive.ts

```diff
@@ -66,14 +61,13 @@ angular.module('oppia').directive('adminMiscTab', [
           ctrl.setStatusMessage('Clearing search index...');
 
           AdminTaskManagerService.startTask();
-          $http.post(ADMIN_HANDLER_URL, {
-            action: 'clear_search_index'
-          }).then(function() {
+          AdminMiscTabBackendApiService.clearSearchIndex()
+            .then(function() {
             ctrl.setStatusMessage('Index successfully cleared.');
             AdminTaskManagerService.finishTask();
-          }, function(errorResponse) {
+          },function(errorResponse) {
             ctrl.setStatusMessage(
-              'Server error: ' + errorResponse.data.error);
+              'Server error: ' + errorResponse.error.error);
```

Comment:
The clearSearchIndex function should return the whole string*

Reply:
Clear Search index is currently returning a whole string.
I have observed that when we make an HTTP request directly through directive it is returning an object with attribute "data" in it.
However, when we make an HTTP request through backend service via promise it is returning htttpErrorResponce with no attribute data init.
That's why i changed "errorResponse.data.error" to "errorResponse.error.error"
And I have tested it manually working fine, And I have checked other files too, it follows the same convection as in our code base structure.